### PR TITLE
Linting and Git ignore

### LINKED ISSUE #1: Setup Linter

### PR DESCRIPTION
**Pull request by wadethestealth**
Puts together a rough outline of some better practices for code style and a git ignore that allows for a environment transferable project.

Note to check linting use `yarn lint`

Fixes #1

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
```
Comment (andrewjmcgehee) [inline .eslintrc]: Can you maybe explain some of these rules to me? Just the gist.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
+node_modules/
+build/
+yarn.lock
```
Comment (andrewjmcgehee) [inline .gitignore]: why remove this? its not removed by default.

File: src/App.js [hunk 3]
```diff
@@ -1,10 +1,10 @@
-import React, { Component } from 'react';
-import Visualizer from './Visualizer'
+import React from 'react';
+import Visualizer from './Visualizer';
 
-export default class App extends Component {
+export default class App extends React.PureComponent {
```
Comment (andrewjmcgehee) [inline src/App.js]: Difference between the two?

File: .gitignore (hunk 2 above)
Comment (wadethestealth) [reply .gitignore]: yarn.lock? because then you cant use npm because they conflict with each other

File: .eslintrc (hunk 1 above)
Comment (wadethestealth) [reply .eslintrc]: basically at the top I add the airbnb config
then 
- allow JSX style code in .js files
- remove line break styling (I think i should set this to unix though)
- enable parameter reassigning
- remove requiring an alt tag (this is about accessiblity and I should probably remove it if we want accessibility)
- use single over double quotes 
- indent (I am going to remove this)
- require comma on last key-value in multiline objects
- max line length 120 character

- some globals eslint needs to know about like window or document

File: src/App.js (hunk 3 above)
Comment (wadethestealth) [reply src/App.js]: Pure Components have no state are are exactly like function components just different syntax. If you just use a render method this speeds up React rendering

File: .eslintrc (hunk 1 above)
Comment (andrewjmcgehee) [reply .eslintrc]: Okay, then:

- [ ] set line endings to unix
- [ ] require alts
- [ ] remove indents

Rest looks fine to me

**Comment by wadethestealth** [comment]
@andrewjmcgehee I have merge access and now setup lint running when you update or open a pull request. If you want to *require passing* in order to merge, you have to setup protected branch status checks I believe.

**Comment by andrewjmcgehee** [comment]
We don't have to "require" it per se. But It should be the expectation that it will pass.

**Pull request merged**